autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -146,32 +151,35 @@ def transform(
 
         if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
             X = typing.cast(pd.DataFrame, X)
-            if np.any(pd.isnull(X)):
-                for column in X.columns:
-                    if X[column].isna().all():
-                        X[column] = pd.to_numeric(X[column])
+            # If we had null columns in our fit call and we made them numeric, then:
+            # - If the columns are null even in transform, apply the same procedure.
+            # - Otherwise, substitute the values with np.NaN and then make the columns numeric.
+            # If the column is null here, but it was not in fit, it does not matter.
+            for column in self.null_columns:
+                # The column is not null, make it null since it was null in fit.
+                if not X[column].isna().all():
```

Comment:
Why do we need this condition?
Because when we create `null_columns`, it requires `X[column].isna().all()`, so isn't it always false?
If that is not the case, we can make flag to avoid this recomputation.

Reply:
this is actually to handle the case where we have all nulls in the training data and some or all non nulls in the testing data for a column. Its actually better to not use this column while testing as we had no information while training. This is based on my discussion with @ArlindKadra. Please let me know if this can be improved.